cypress/e2e/crypto/decryption-failure.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { UserCredentials } from "../../support/login";
+import Chainable = Cypress.Chainable;
+
+const ROOM_NAME = "Test room";
+const TEST_USER = "Alia";
+const BOT_USER = "Benjamin";
+
+type EmojiMapping = [emoji: string, name: string];
+
+const waitForVerificationRequest = (cli: MatrixClient): Promise<VerificationRequest> => {
+    return new Promise<VerificationRequest>(resolve => {
+        const onVerificationRequestEvent = (request: VerificationRequest) => {
+            // @ts-ignore CryptoEvent is not exported to window.matrixcs; using the string value here
+            cli.off("crypto.verification.request", onVerificationRequestEvent);
+            resolve(request);
+        };
+        // @ts-ignore
+        cli.on("crypto.verification.request", onVerificationRequestEvent);
+    });
+};
+
+const handleVerificationRequest = (request: VerificationRequest): Chainable<EmojiMapping[]> => {
+    return cy.wrap(new Promise<EmojiMapping[]>((resolve) => {
+        const onShowSas = (event: ISasEvent) => {
+            verifier.off("show_sas", onShowSas);
+            event.confirm();
+            resolve(event.sas.emoji);
+        };
+
+        const verifier = request.beginKeyVerification("m.sas.v1");
+        verifier.on("show_sas", onShowSas);
+        verifier.verify();
+    }));
+};
+
+describe("Decryption Failures", () => {
+    let synapse: SynapseInstance | undefined;
+    let testUser: UserCredentials | undefined;
+    let bot: MatrixClient | undefined;
+    let roomId: string;
+
+    beforeEach(function() {
+        cy.startSynapse("default").then((syn: SynapseInstance) => {
+            synapse = syn;
+            cy.initTestUser(synapse, TEST_USER).then((creds: UserCredentials) => {
+                testUser = creds;
+            }).then(() => {
+                cy.getBot(synapse, { displayName: BOT_USER }).then((cli) => {
+                    bot = cli;
+                });
+            }).then(() => {
+                cy.createRoom({ name: ROOM_NAME }).then((id) => {
+                    roomId = id;
+                });
+            }).then(() => {
+                cy.inviteUser(roomId, bot.getUserId());
+                cy.visit("/#/room/" + roomId);
+                cy.contains(".mx_TextualEvent", BOT_USER + " joined the room").should("exist");
+            }).then(() => {
+                cy.getClient().then(async (cli) => {
+                    await cli.setRoomEncryption(roomId, { algorithm: "m.megolm.v1.aes-sha2" });
+                    await bot.setRoomEncryption(roomId, { algorithm: "m.megolm.v1.aes-sha2" });
+                }).then(() => {
+                    bot.getRoom(roomId).setBlacklistUnverifiedDevices(true);
+                });
+            });
+        });
+    });
+
+    afterEach(() => {
+        cy.stopSynapse(synapse);
+    });
+
+    it("the decryption failure bar should prompt the user to verify, "
+        + "if this device isn't verified and there are other verified devices or backups", () => {
```

Comment:
these test descriptions don't really compose very well with the `describe` text.

Maybe

```
describe("The decryption failure bar", () => {
   it("should prompt the user to verify ... etc", ...
```